### PR TITLE
godap 2.5.0 (new formula)

### PR DESCRIPTION
**Pull request by harpchad**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/g/godap.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  homepage "https://github.com/Macmod/godap"
+  url "https://github.com/Macmod/godap.git",
+    tag:      "v2.5.0",
+    revision: "a5201bc2426d2fbeb469b40c532d1974246d487e"
+  license "MIT"
+
+  livecheck do
+    url :stable
+  end
+
```
Comment (chenrui333) [inline Formula/g/godap.rb:12]:
```suggestion
```

File: Formula/g/godap.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w -X main.Version=#{version}")
+    generate_completions_from_executable(bin/"godap",  "completion")
+  end
+
+  test do
+    assert_includes shell_output("#{bin}/godap -T 1 203.0.113.1 2>&1", 1),
+      "LDAP Result Code 200 \"Network Error\": dial tcp 203.0.113.1:389: i/o timeout"
```
Comment (chenrui333) [inline Formula/g/godap.rb:22]:
```suggestion
    output = shell_output("#{bin}/godap -T 1 203.0.113.1 2>&1", 1)
    assert_equal "LDAP Result Code 200 \"Network Error\": dial tcp 203.0.113.1:389: i/o timeout", output

    assert_match version.to_s, shell_output("#{bin}/godap version")
```

Comment (harpchad) [reply Formula/g/godap.rb:17]: The output includes a timestamp:
2024/05/14 09:46:30 LDAP Result Code 200 "Network Error": dial tcp 203.0.113.1:389: i/o timeout

How about:
```golang
-    assert_includes shell_output("#{bin}/godap -T 1 203.0.113.1 2>&1", 1),
-      "LDAP Result Code 200 \"Network Error\": dial tcp 203.0.113.1:389: i/o timeout"
+    output = shell_output("#{bin}/godap -T 1 203.0.113.1 2>&1", 1)
+    assert_includes output, "LDAP Result Code 200 \"Network Error\": dial tcp 203.0.113.1:389: i/o timeout"
+
+    assert_match version.to_s, shell_output("#{bin}/godap version")
```

Comment (chenrui333) [reply Formula/g/godap.rb]: yeah, indeed, using `assert_match` should cover it.

**Comment by chenrui333** [comment]
@harpchad can you also include this formula into autobump.txt and squash all commits into one? Thanks!

File: Formula/g/godap.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w -X main.Version=#{version}")
+    generate_completions_from_executable(bin/"godap",  "completion")
+  end
+
+  test do
+    output = shell_output("#{bin}/godap -T 1 203.0.113.1 2>&1", 1)
+    assert_includes output, "LDAP Result Code 200 \"Network Error\": dial tcp 203.0.113.1:389: i/o timeout"
```
Comment (chenrui333) [inline Formula/g/godap.rb:17]: this should be good

```suggestion
    assert_match "LDAP Result Code 200 \"Network Error\": dial tcp 203.0.113.1:389: i/o timeout", output
```

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@harpchad, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**